ignite/pkg/cosmosgen/generate_typescript.go

```diff
@@ -0,0 +1,179 @@
+package cosmosgen
+
+import (
+	"github.com/ignite-hq/cli/ignite/pkg/cosmosanalysis/module"
+	"github.com/ignite-hq/cli/ignite/pkg/giturl"
+	"github.com/ignite-hq/cli/ignite/pkg/gomodulepath"
+	"github.com/ignite-hq/cli/ignite/pkg/nodetime/programs/sta"
+	tsproto "github.com/ignite-hq/cli/ignite/pkg/nodetime/programs/ts-proto"
+	"github.com/ignite-hq/cli/ignite/pkg/protoc"
+
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	tsOut = []string{
+		"--ts_proto_out=.",
+	}
+
+	jsOpenAPIOut = []string{
+		"--openapiv2_out=logtostderr=true,allow_merge=true,json_names_for_fields=false,Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:.",
+	}
+)
+
+type tsGenerator struct {
+	g *generator
+}
+
+type generatePayload struct {
+	Modules []module.Module
+	User    string ``
+	Repo    string ``
+}
+
+func newTSGenerator(g *generator) *tsGenerator {
+	return &tsGenerator{
+		g: g,
+	}
+}
+
+func (g *generator) generateTS() error {
+	tsg := newTSGenerator(g)
+
+	chainPath, _, err := gomodulepath.Find(g.appPath)
+	if err != nil {
+		return err
+	}
+
+	chainInfo, err := giturl.Parse(chainPath.RawPath)
```

Comment:
We need to refactor the "user/repo" related logic to use the app module path to define the name(s) that will be used later on for the npm package names. Since #2353 we deprecated the "user/repo" references and the `giturl` package because it was not being used anymore.